From d386b3f30ca78f10bbecaf72a00a63775eab2110 Mon Sep 17 00:00:00 2001
From: Thomas D.
Date: Wed, 11 Mar 2015 00:03:17 +0100
Subject: [PATCH] configure: Enable valgrind only when explicit declared

rsyslog should only use valgrind when requested with "--enable-valgrind".
This allows you to disable valgrind usage and guessing is always bad.
---
 configure.ac | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0c72f80..c94be0d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,10 +13,6 @@ AC_CONFIG_HEADERS([config.h])
 
 AC_USE_SYSTEM_EXTENSIONS
 
-# check if valgrind is present
-AC_CHECK_PROG(have_valgrind, [valgrind], [yes])
-AM_CONDITIONAL(HAVE_VALGRIND, test x$have_valgrind = xyes)
-
 # Checks for programs.
 AC_PROG_LEX
 AC_PROG_YACC
@@ -496,8 +492,15 @@ AC_ARG_ENABLE(valgrind,
         [enable_valgrind="no"]
 )
 if test "$enable_valgrind" = "yes"; then
-        AC_DEFINE(VALGRIND, 1, [Defined if valgrind support settings are to be enabled (e.g. prevents dlclose()).])
+        AC_CHECK_PROG(VALGRIND, [valgrind], [valgrind], [no])
+
+        AS_IF([test "x$VALGRIND" = "xno"], [
+                AC_MSG_ERROR([valgrind is missing but enabled. Install valgrind or do not enable valgrind in configure])
+        ], [
+                AC_DEFINE(VALGRIND, 1, [Defined if valgrind support settings are to be enabled (e.g. prevents dlclose()).])
+        ])
 fi
+AM_CONDITIONAL([HAVE_VALGRIND], [test "$enable_valgrind" = "yes" && test "x$VALGRIND" != "xno"])
 
 
 # memcheck
-- 
2.3.1

